doc/sg/c_cpp_coding_style.md

```diff
@@ -323,6 +369,13 @@ Macros are often necessary and reasonable coding practice C (as opposed to C++)
 In contrast to the recommendation in the Google C++ style guide, exporting macros as part of the public API is allowed in C code.
 A typical use case is a header with register definitions.
 
+**Function-line macros**
+Function-like macros should be avoided whenever is possible since it is costly to maintain. Where it is necessary, it should be hygienic, here are some useful tips:
```

Comment:
```suggestion
Function-like macros should be avoided whenever possible since they are error-prone. Where they are necessary, they should be hygienic. Here are some useful tips:
```